web/pages/Settings/AISettings.tsx

```diff
@@ -0,0 +1,99 @@
+import { adaptersToOptions } from '../../shared/util'
+import Select from '../../shared/Select'
+import {
+  ADAPTER_LABELS,
+} from '../../../common/adapters'
+import { settingStore, userStore } from '../../store'
+import Tabs from '../../shared/Tabs'
+import { AIPreset } from './components/AIPreset'
+import HordeAISettings from './components/HordeAISettings'
+import { Component, Show, createEffect, createMemo, createSignal } from 'solid-js'
+import OpenAISettings from './components/OpenAISettings'
+import ScaleSettings from './components/ScaleSettings'
+import NovelAISettings from './components/NovelAISettings'
+import LuminAISettings from './components/LuminAISettings'
+import KoboldAISettings from './components/KoboldAISettings'
+import ChaiSettings from './components/ChaiSettings'
+
+const AISettings: Component<{
+  onHordeWorkersChange: (workers: string[]) => void
+}> = (props) => {
+  const state = userStore()
+  const cfg = settingStore()
+
+  createEffect(() => {
+    setTabs(cfg.config.adapters.map(a => ADAPTER_LABELS[a] || a))
+    if(!ready() && cfg.config.adapters?.length) {
+      var defaultTab = (cfg.config.adapters as (string | undefined)[]).indexOf(state.user?.defaultAdapter)
+      setTab(defaultTab == -1 ? 0 : defaultTab)
+      setReady(true)
+    }
+  })
+
+  const [tabs, setTabs] = createSignal<string[]>([])
+  const [tab, setTab] = createSignal(-1)
+  const [ready, setReady] = createSignal(false)
+
+  const currentTab = createMemo(() => cfg.config.adapters[tab()])
+
+  const tabClass = `flex flex-col gap-4`
+
+  return (
+    <>
+      <Show when={!ready()}>
+        <div>Loading...</div>
+      </Show>
+
+      <Show when={ready()}>
+        <Select
+          fieldName="defaultAdapter"
+          label="Default AI Service"
+          items={adaptersToOptions(cfg.config.adapters)}
+          helperText="The default service conversations will use unless otherwise configured"
+          value={state.user?.defaultAdapter}
+        />
+
+        <div class="my-2">
+          <Tabs tabs={tabs()} selected={tab} select={setTab} />
+        </div>
+      </Show>
+
+      <div class={currentTab() === 'horde' ? tabClass : 'hidden'}>
```

Comment:
This looks great on desktop, but need to consider mobile as well.

Reply:
To me it worked well, although it needs scrolling, but it was this way before, too. If/when I work on presets and moving those in a separate Backend menu, maybe there's a different UI better suited for this, such as a list that brings you to a dedicated per-backend page instead of tabs (like the chats list that brins you to the chat details).